### PR TITLE
Validate complete order request

### PR DESCRIPTION
**Pull request by kayue**
The first step to fix https://github.com/Sylius/ShopApiPlugin/issues/598 is to perform validation in `CompleteOrderAction`

### CODE REVIEW

**Comment by mamazu** [comment]
What does it validate at the moment, if you haven't have any validation specified.

**Comment by kayue** [comment]
@mamazu Ideally it should re-validate inventory availability, which can be implemented later. 

Nevertheless, Sylius is meant to be extendable, even the ShopAPI doesn't have any validation specified, it should still perform validation to allower developer to add custom validator.

**Comment by kayue** [comment]
Also due to inconsistent implementation in action classes, error is being return in multiple different formats. 

For example, invalid cart token, it returns proper error 400 in `PutItemToCartAction`:

```
{
  "code": 400,
  "message": "Validation failed",
  "errors": {
    "token": [
      "Cart does not exist."
    ]
  }
}
```

However, in `CompleteOrderAction` it just return error 500 due to lack of validation:

```
{
  "code": 500,
  "message": "Order with ca5ac343-f09d-4cf0-8d34-374d77cd2ca91 token has not been found."
}
```

In `SummarizeAction` it returns error 404

```
{
  "code": 404,
  "message": "Cart with given id does not exists"
}
```

**Comment by mamazu** [comment]
Currently as you found the order complete doesn't validate anything and therefore only asserts that the cart is there (which then will cause an error => 500) We should definitely add the same validators as the other endpoints have as well.

**Comment by lchrusciel** [comment]
Hey @kayue,

are you willing to add test for this case?

File: spec/Validator/Cart/CartReadyForCheckoutValidatorSpec.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        $this->validate('CART_TOKEN', new CartReadyForCheckout());
+    }
+
+    function it_adds_a_violation_if_cart_can_not_checkout(
+        RepositoryInterface $repository,
+        OrderInterface $order,
+        FactoryInterface $stateMachineFactory,
+        StateMachineInterface $stateMachine,
+    ExecutionContextInterface $context
```
Comment (lchrusciel) [inline spec/Validator/Cart/CartReadyForCheckoutValidatorSpec.php]:
```suggestion
        ExecutionContextInterface $context
```

**Comment by lchrusciel** [comment]
Thanks, Ka! :1st_place_medal:

**Comment by kayue** [comment]
Thanks @lchrusciel for adding the tests. I didn't know what to test but now I learned.

**Comment by lchrusciel** [comment]
All credits to @mamazu. I just asked him for help ;)

**Pull request merged**